darts/tests/models/components/test_layer_norm_variants.py

```diff
@@ -25,3 +28,37 @@ def test_lnv(self):
                 ln = layer_norm(4)
                 inputs = torch.zeros(1, 4, 4)
                 ln(inputs)
+
+        def test_rin(self):
+
+            x = torch.randn(3, 4, 7)
+
+            affine_options = [True, False]
+            axis_options = [1, 2, -1, -2]
+            input_dim_options = list(x.shape[1:])
+
+            test_runs = itertools.product(
+                affine_options, axis_options, input_dim_options
+            )
+
+            for run in test_runs:
+
+                affine, axis, input_dim = run
+
+                if (x.shape[axis] is input_dim) and affine:
+
+                    with self.assertRaises(RuntimeError):
+                        rin = ReversibleInstanceNorm(
+                            axis=axis, input_dim=input_dim, affine=affine
+                        )
+                        x_norm = rin(x, "norm")
+                        x_denorm = rin(x_norm, "denorm")
```

Comment:
it's better to only have the line which should raise the Error inside the with body.
The Norm creation should not raise an error. Will the error always be raised in the `x_norm` line?

Then we could rewrite as:
```suggestion
                    rin = ReversibleInstanceNorm(
                        axis=axis, input_dim=input_dim, affine=affine
                    )
                    with self.assertRaises(RuntimeError):
                        _ = rin(x, "norm")
                    with self.assertRaises(RuntimeError):
                        _ = rin(x, "denorm")
```